File: backend/repositories/auth_repository.py
# backend/repositories/auth_repository.py
from typing import Optional, Tuple

from repositories.base_repository import BaseRepository, DatabaseConnectionError, QueryError

AUTH_TABLE = "USW_DAT"


class AuthRepositoryError(Exception):
    """Raised when the auth repository encounters an error."""
    pass


class AuthRepository(BaseRepository):
    """Repository for user authentication data (table USW_DAT)."""

    def find_user_by_login(self, login: str) -> Optional[Tuple[str, str]]:
        """
        Return (USW_LOGN, USW_PASS) for the given login, or None if not found.

        Raises:
            AuthRepositoryError: on DB connection or query failure.
        """
        query = f"SELECT USW_LOGN, USW_PASS FROM {AUTH_TABLE} WHERE USW_LOGN = ?"
        try:
            rows = self.execute_query(query, (login,))
        except (DatabaseConnectionError, QueryError) as exc:
            raise AuthRepositoryError(str(exc)) from exc

        if not rows:
            return None

        row = rows[0]
        return (row[0], row[1])

File: backend-go/repositories/auth_repository.go
package repositories

import (
	"database/sql"
	"errors"
	"fmt"
)

// AuthUser holds the credentials read from the USW_DAT table.
type AuthUser struct {
	Login    string
	Password string
}

// AuthRepository provides access to user authentication data (table USW_DAT).
type AuthRepository struct {
	DB *sql.DB
}

// NewAuthRepository wires the shared connection pool into the repository.
func NewAuthRepository(db *sql.DB) *AuthRepository {
	return &AuthRepository{DB: db}
}

// FindUserByLogin returns the user matching the given login, or (nil, nil) when
// no such user exists. Any database/query failure is returned as an error.
func (r *AuthRepository) FindUserByLogin(login string) (*AuthUser, error) {
	const query = "SELECT USW_LOGN, USW_PASS FROM USW_DAT WHERE USW_LOGN = @p1"

	row := r.DB.QueryRow(query, login)

	var user AuthUser
	if err := row.Scan(&user.Login, &user.Password); err != nil {
		if errors.Is(err, sql.ErrNoRows) {
			return nil, nil
		}
		return nil, fmt.Errorf("find user by login failed: %w", err)
	}

	return &user, nil
}
